### PR TITLE
Allow `user.followed` field to be sorted by recency of content

### PR DESCRIPTION
**Pull request by laurieboyes**
Enables cool queries like this one, which will be useful in myFT page and beyond:
```
{
  user(uuid: "f0bb6f11-c034-4792-a1c0-3647825363a0") {
    uuid,
    followed(sortBy: MostRecentContent) {
      name
    }
  }
}
```

Not sure whether it's controversial to make extra capi calls for sorting the concepts like this. Alternative would be to sort after making this query, but it seemed neater and more expressive to wrap this into the API.

### CODE REVIEW

File: server/graphql/types/user.js [hunk 1]
```diff
@@ -11,6 +12,46 @@ function getMyFtTags (contentTags, followedTags) {
...
+			description: 'No sorting, however comes from the myFT DB'
+		},
+		MostRecentContent: {
+			value: 'mostRecentContent',
+			description: 'Sort concepts by how recently they had content'
+		}
+	}
+});
+
+function getConceptLastPublished (conceptId, backends) {
```
Comment (ironsidevsquincy) [inline server/graphql/types/user.js]: took me a few reads to understand this; maybe `getContceptLastUsed`?

File: server/graphql/types/user.js [hunk 2]
```diff
@@ -11,6 +12,46 @@ function getMyFtTags (contentTags, followedTags) {
...
+	return backends.capi.search({ term: { 'metadata.idV1': conceptId } }, { limit: 1 })
+		.then(([latestContentItem]) => latestContentItem.publishedDate);
+}
+
+function getConceptsWithLastPublished (concepts, backends) {
+	return Promise.all(concepts.map(concept => getConceptLastPublished(concept.id, backends)
+		.then(contentLastPublished => Object.assign({
+			contentLastPublished
```
Comment (ironsidevsquincy) [inline server/graphql/types/user.js]: `...LastUsed`?

File: server/graphql/types/user.js [hunk 3]
```diff
@@ -11,6 +12,46 @@ function getMyFtTags (contentTags, followedTags) {
...
+		}, concept)))
+	)
+}
+
+function sortConceptsByMostRecentContent (concepts, backends) {
+	return getConceptsWithLastPublished(concepts, backends)
+		.then(conceptsWithLastPublished => {
+			return conceptsWithLastPublished.sort((a, b) => {
+				const aTime = moment(a.contentLastPublished).valueOf();
+				const bTime = moment(b.contentLastPublished).valueOf();
```
Comment (ironsidevsquincy) [inline server/graphql/types/user.js]: can't you use `new Date(...)`?

File: server/graphql/types/user.js (hunk 1 above)
Comment (laurieboyes) [inline server/graphql/types/user.js]: How about `getConceptLastHadContent`?

Comment (ironsidevsquincy) [inline server/graphql/types/user.js]: yep, or that

**Pull request merged**